lib/travis/hub/support/amqp.rb

```diff
@@ -4,24 +4,25 @@
 module Travis
   class Amqp
     class << self
-      def setup(config)
-        @instance ||= new(config).tap do |amqp|
+      def setup(config, enterprise = false)
+        @instance ||= new(config, enterprise).tap do |amqp|
           amqp.setup
         end
       end
     end
 
     attr_reader :connection, :channel
 
-    def initialize(config)
+    def initialize(config, enterprise = false)
       @connection = Bunny.new(config.to_h).tap { |connection| connection.start }
       # TODO channels must not be shared across threads, set this to Thread.current?
       @channel = connection.create_channel
+      @enterprise = enterprise
     end
 
     def setup
       # TODO required on enterprise. move details to config?
-      if Travis.config.enterprise?
+      if @enterprise
```

Comment:
Do we need this to be in a config? Or is it ok to run in production? For now I'm passing in if we're running in enterprise since the context isn't available calling `Hub.config` is deprecated.